### PR TITLE
Fix page number wrong in sidepanel

### PR DESCRIPTION
**Pull request by cdrini**
The numbers were incorrect for some reason. Change them to be consistent with the numbers on the search pins. Still not fully consistent with e.g. slider, but we can solve that another day. They are at least not wildly incorrect now.

| Before | After |
| --- | --- |
| image | image |

### CODE REVIEW

File: src/BookNavigator/search/a-search-result.js [hunk 1]
```diff
@@ -35,17 +36,17 @@ export class BookSearchResult extends LitElement {
...
   render() {
-    const { match } = this;
-    const { par = [] } = match;
-    const [resultDetails = {}] = par;
-    const pageNumber = Number.isInteger(resultDetails.page)
-      ? html`<p class="page-num">Page -${resultDetails.page}-</p>` : nothing;
+    /** @type {SearchInsideMatch} */
+    const match = this.match;
+    // TODO: Make this not use the global br instance
+    const pageIndex = window.br.leafNumToIndex(match.par[0].page);
```
Comment (iisa) [inline src/BookNavigator/search/a-search-result.js:42]: hi @cdrini thanks for this! can you pipe it thru via the `searchProvider`? that layer receives `br` instance, and we won't have to call global var..

Comment (iisa) [inline src/BookNavigator/search/a-search-result.js:42]: hi @cdrini thanks for this! can you pipe it thru via the `searchProvider`? that layer receives `br` instance, and we won't have to call global var..

Comment (cdrini) [reply src/BookNavigator/search/a-search-result.js]: I think I found a good fix! Let me know what you think :)

File: src/plugins/search/plugin.search.js [hunk 2]
```diff
@@ -269,22 +272,32 @@ BookReader.prototype.cancelSearchRequest = function () {
...
- * @param {boolean} options.goToFirstResult
+ * @param {(pageNum: LeafNum) => PageNumString} displayPageNumberFn
  */
-BookReader.prototype.BRSearchCallback = function(results, options) {
+export function marshallSearchResults(results, displayPageNumberFn) {
   // Attach matchIndex to a few things to make it easier to identify
   // an active/selected match
   for (const [index, match] of results.matches.entries()) {
     match.matchIndex = index;
+    match.displayPageNumber = displayPageNumberFn(match.par[0].page);
```
Comment (iisa) [inline src/plugins/search/plugin.search.js:284]: ok. I was hoping that the service would return it with the result sooner than later 😅

**Review by iisa — APPROVED** [review]
code looks good!
tested here: https://deploy-preview-1011--lucid-poitras-9a1249.netlify.app/bookreaderdemo/demo-internetarchive?ocaid=goody#page/10/mode/2up/search/goody

I tried searching 'the' and it took a reasonable amount of time to draw 👍 
 https://deploy-preview-1011--lucid-poitras-9a1249.netlify.app/bookreaderdemo/demo-internetarchive?ocaid=theworksofplato01platiala#page/430/mode/2up/search/the

weird that the Plato "the" search did not display results highlights properly. perhaps it's data bc `goody` search above displayed the highlights well.

**Comment by cdrini** [comment]
Yeah :/ I think there's a JIRA ticket for that issue: https://webarchive.jira.com/browse/WEBDEV-5190

**Pull request merged**